solutions/ha-nva/layer-7-ingress/azuredeploy.json

```diff
@@ -0,0 +1,801 @@
+                "addressSpace": {
+                    "addressPrefixes": [
+                        "10.0.0.0/16"
+                    ]
+                },
+                "subnets": [
+                    {
+                        "name": "GatewaySubnet",
+                        "properties": {
+                            "addressPrefix": "10.0.255.224/27"
+                        }
+                    },
+                    {
+                        "name": "ApplicationGatewaySubnet",
+                        "properties": {
+                            "addressPrefix": "10.0.252.224/27",
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('DmzNetworkSecurityGroupName'))]"
+                            }
+                        }
+                    },
+                    {
+                        "name": "AzureBastionSubnet",
+                        "properties": {
+                            "addressPrefix": "10.0.253.224/27",
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('DmzNetworkSecurityGroupName'))]"
+                            }
+                        }
+                    },
+                    {
+                        "name": "snet-dmz",
+                        "properties": {
+                            "addressPrefix": "10.0.0.64/27",
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('DmzNetworkSecurityGroupName'))]"
+                            }
+                        }
+                    },
+                    {
+                        "name": "snet-web",
+                        "properties": {
+                            "addressPrefix": "10.0.1.0/24",
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('WebNetworkSecurityGroupName'))]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "condition": "[equals(parameters('DeployAzureBastionDmz'), 'Yes')]",
+            "type": "Microsoft.Network/publicIPAddresses",
+            "apiVersion": "2019-12-01",
+            "name": "pip-bastion-hanva",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "Standard"
+            },
+            "properties": {
+                "publicIPAllocationMethod": "Static"
+            }
+        },
+        {
+            "condition": "[equals(parameters('DeployAppGatewayDmz'), 'Yes')]",
+            "type": "Microsoft.Network/publicIPAddresses",
+            "apiVersion": "2019-12-01",
+            "name": "pip-appgateway-hanva",
```

Comment:
Probably a discussion to be had around hard coding names vs. using parameters with default values. I've been adding objet typed parameters with all configurable values. It's not super convenient when specifying new values but does leave the option while keeping the number of parameters minimal.

example: https://github.com/mspnp/samples/blob/master/solutions/basic-web-app/azuredeploy.json#L27

Reply:
I'm a bit conflicted, and I know I have a mix in this sample.  I tend to think that the "Deploy to Azure" experience should be user-friendly so that the customer doesn't feel like they have to make a bunch of decisions if they just want to deploy and check out the architecture. Putting a bunch of params into a single object param might be confusing in the UI when they see the default value is a big json blob.  One other idea to consider:  Place all the things that an expert user might want to change, but someone just kicking the tires doesn't need to worry about as values in variables?  That raises the bar a bit for someone to take a sample architecture and incorporate it into their own environment, but it might make things more approachable for the customer that just wants to give it a try. I can see good reasons to go in each direction. We should figure out a good standard and try to incorporate it into our best practices.

Reply:
I agree. From a scope perspective the issue is how to provide a learning experience for the learners, and a production readiness experience for architects. I intentionally stayed away from variables to provide a flexible deployment experience, but totally agree that it's overwhelming for the learner.

Let's figure it out.